docs/xks/developer-guide/scheduling.md

```diff
@@ -0,0 +1,122 @@
+3. You can constrain against labels on other pods running on the node (or other topological domain), rather than against labels on the node itself.
+
+### Node Affinity
+
+There are currently two types of node affinity, called "requiredDuringSchedulingIgnoredDuringExecution" and "preferredDuringSchedulingIgnoredDuringExecution". You can think of them as "hard" and "soft" requirements to schedule a pod. "IgnoredDuringExecution" part of the names means that, if labels on a node change at runtime such that the affinity rules on a pod are no longer met, the pod continues to run on the node.
+
+
+
+```yaml
+spec:
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: kubernetes.io/name
+            operator: In
+            values:
+            - ABC
+            - XYZ
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        preference:
+          matchExpressions:
+          - key: label-key
+            operator: In
+            values:
+            - label-value
+```
+
+This example only allow pods to be scheduled on nodes with a key "kubernetes.io/name" with value "ABC" or "XYZ". Among the nodes matching this criteria, nodes with key "label-key" and value "label-value" will be preferred.
+
+The "weight" field ir ranged 1-100 and for each node matching all scheduling requirements, the kube-scheduler computes a score, as mentioned earlier. Then adds this number to that sum to calculate the best matching node.
+
+### podAffinity and podAntiaffinity
+
+podAffinity and podAntiAffinity lets you constrain which nodes pod are eligible to be scheduled on based of label of the pods running on the node rather then labels on the node. 
+
+podAnitAffinity
+
+```yaml
+spec:
+  affinity:
+    podAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+          - key: label1
+            operator: In
+            values:
+            - label-value
+        topologyKey: topology.kubernetes.io/zone
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: label2
+              operator: In
+              values:
+              - label-value-anti
+          topologyKey: topology.kubernetes.io/zone
+```
+This example show an example where we use both affinity rules.
+
+Affinity rule: the pod can be only scheduled onto a node if that node is in the same zone as at least one already-running pod that has a label with key "label1" and value "label-value".
+
+antiAffinity rule: the pod should not be scheduled onto a node if that node is in the same zone as a pod with label having key "label2" and value "label-valu-anti"
+
+```yaml

```

Comment:
https://github.com/XenitAB/terraform-modules/blob/e6df36b0dc7d3573bba6b04ecdf4407948d7afe0/modules/kubernetes/prometheus/charts/prometheus-extras/templates/prometheus.yaml#L90-L110

Please use this example.